src/Ast.ml

```diff
@@ -44,6 +44,23 @@ let is_prefix exp =
   | Pexp_ident {txt= Lident i; _} -> is_prefix_id i
   | _ -> false
 
+let is_infix_symbol = function
+  | '$' | '&' | '*' | '+' | '-' | '/' | '<' | '=' | '>' | '@' | '^' | '|' ->
```

Comment:
I think `is_infix_symbol` should be renamed to something more specific (is_kwdopchar ?). Also, I don't think this list is accurate. `!`, `%`, `:`, `?` seem to be missing.

Reply:
It should consider the union of the following two definition:
```
let dotsymbolchar =
  ['!' '$' '%' '&' '*' '+' '-' '/' ':' '=' '>' '?' '@' '^' '|']
let kwdopchar =
  ['$' '&' '*' '+' '-' '/' '<' '=' '>' '@' '^' '|']
```

Reply:
> I think `is_infix_symbol` should be renamed to something more specific (is_kwdopchar ?). Also, I don't think this list is accurate. `!`, `%`, `:`, `?` seem to be missing.

This list comes from the ocaml PR: https://github.com/ocaml/ocaml/pull/1947/files#diff-635f3857747d984a68e56f2afdc27a4b

Reply:
Good catch, fixed.